ogr/ogrsf_frmts/hana/ogrhanalayer.cpp

```diff
@@ -0,0 +1,955 @@
+        sql = StringFormat(
+            "SELECT ext.ST_XMin(),ext.ST_YMin(),ext.ST_XMax(),ext.ST_YMax() "
+            "FROM (%s)",
+            subQuery.c_str());
+    }
+
+    extent->MinX = 0.0;
+    extent->MaxX = 0.0;
+    extent->MinY = 0.0;
+    extent->MaxY = 0.0;
+
+    odbc::StatementRef stmt = dataSource_->CreateStatement();
+    odbc::ResultSetRef rsExtent = stmt->executeQuery(sql.c_str());
+    if (rsExtent->next())
+    {
+        odbc::Double val = rsExtent->getDouble(1);
+        if (!val.isNull())
+        {
+            extent->MinX = *val;
+            extent->MinY = *rsExtent->getDouble(2);
+            extent->MaxX = *rsExtent->getDouble(3);
+            extent->MaxY = *rsExtent->getDouble(4);
+        }
+    }
+    rsExtent->close();
+}
+
+/************************************************************************/
+/*                          ResetReading()                              */
+/************************************************************************/
+
+void OGRHanaLayer::ResetReading()
+{
+    nextFeatureId_ = 0;
+    BuildQueryStatement();
+}
+
+/************************************************************************/
+/*                            GetExtent()                               */
+/************************************************************************/
+
+OGRErr OGRHanaLayer::GetExtent(int iGeomField, OGREnvelope* extent, int force)
+{
+    if( iGeomField < 0 || iGeomField >= GetLayerDefn()->GetGeomFieldCount() ||
+        GetLayerDefn()->GetGeomFieldDefn(iGeomField)->GetType() == wkbNone )
+    {
+        extent->MinX = 0.0;
+        extent->MaxX = 0.0;
+        extent->MinY = 0.0;
+        extent->MaxY = 0.0;
+
+        if( iGeomField != 0 )
+        {
+            CPLError(CE_Failure, CPLE_AppDefined,
+                     "Invalid geometry field index : %d", iGeomField);
+        }
+        return OGRERR_FAILURE;
+    }
+
+    try
+    {
+        ReadGeometryExtent(iGeomField, extent);
+        return OGRERR_NONE;
+    }
+    catch (const std::exception& ex)
+    {
+        CPLString clmName =
+            (iGeomField < static_cast<int>(geomColumns_.size()))
+                ? geomColumns_[static_cast<std::size_t>(
+                                   geomColumns_.size())].name
+                : "unknown column";
+        CPLError(CE_Failure, CPLE_AppDefined,
+            "Unable to query extent of '%s' using fast method: %s",
+            clmName.c_str(), ex.what());
+    }
+
+    if( iGeomField == 0 )
+        return OGRLayer::GetExtent( extent, force );
+    else
+        return OGRLayer::GetExtent( iGeomField, extent, force);
+}
+
+/************************************************************************/
+/*                            GetFeatureCount()                          */
+/************************************************************************/
+
+GIntBig OGRHanaLayer::GetFeatureCount(CPL_UNUSED int force)
+{
+    GIntBig ret = FALSE;
```

Comment:
```suggestion
    GIntBig ret = 0;
```

Reply:
Fixed in 644e57d53b752d4ef164e70e9424b3bc73246580.